runtime/queries/regex/highlights.scm

```diff
@@ -50,4 +50,3 @@
   ])
 
 (class_character) @constant.character
-(pattern_character) @string
```

Comment:
Can we add the `(ERROR)` class so that invalid regexes are highlighted? That's something that has been a TODO in the prompt for a while, the idea was to highlight the whole line as red but I guess we can actually do better and use `(ERROR)` nodes

Reply:
(This TODO: https://github.com/helix-editor/helix/blob/fb103eef3ef448c0983e1534d1a5dfae405c0deb/helix-term/src/ui/mod.rs#L147 kakoune just colors the whole line as red)

Reply:
I think we could overlay some error highlights for regex syntax errors according to rust's regex without too much extra work. We could check the syntax with the `regex-syntax` dep and using the [location span](https://docs.rs/regex-syntax/0.7.4/regex_syntax/ast/struct.Error.html#method.span) in a syntax error, then merging those as the `additional_highlight_spans`: https://github.com/helix-editor/helix/blob/5a5289701429c2d7019d11d2184833745865063f/helix-term/src/ui/markdown.rs#L29-L35

Reply:
Ah that's a really neat idea 👍 Although you need to be sure to convert byte/utf8 indeces to char/utf32 indices